core/meterer/onchain_state.go

```diff
@@ -106,63 +111,84 @@ func (pcs *OnchainPaymentState) RefreshOnchainPaymentState(ctx context.Context,
 		return err
 	}
 	// These parameters should be rarely updated, but we refresh them anyway
-	pcs.PaymentVaultParams = paymentVaultParams
+	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&pcs.PaymentVaultParams)), unsafe.Pointer(paymentVaultParams))
```

Comment:
Do we need to use unsafe pointer? Can we just type `PaymentVaultParams` as `atomic.Pointer[PaymentVaultParams]` in L41?

Reply:
aah cool! updated